src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyScan.java

```diff
@@ -0,0 +1,292 @@
+import static com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures.OWASP_DEPENDENCY_CHECK_USAGE;
+import static com.sap.sgs.phosphor.fosstars.model.other.Utils.setOf;
+import static com.sap.sgs.phosphor.fosstars.model.value.OwaspDependencyCheckUsage.MANDATORY;
+import static com.sap.sgs.phosphor.fosstars.model.value.OwaspDependencyCheckUsage.NOT_USED;
+import static com.sap.sgs.phosphor.fosstars.model.value.OwaspDependencyCheckUsage.OPTIONAL;
+
+import com.sap.sgs.phosphor.fosstars.maven.ModelVisitor;
+import com.sap.sgs.phosphor.fosstars.maven.ModelVisitor.Location;
+import com.sap.sgs.phosphor.fosstars.model.Feature;
+import com.sap.sgs.phosphor.fosstars.model.ValueSet;
+import com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures;
+import com.sap.sgs.phosphor.fosstars.model.value.OwaspDependencyCheckUsage;
+import com.sap.sgs.phosphor.fosstars.model.value.ValueHashSet;
+import com.sap.sgs.phosphor.fosstars.tool.github.GitHubProject;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.model.ReportPlugin;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+
+/**
+ * This data provider checks if an open-source project uses OWASP Dependency Check to scan
+ * dependencies for known vulnerabilities. In particular, it tires to fill out the following
+ * features:
+ * <ul>
+ *   <li>{@link OssFeatures#OWASP_DEPENDENCY_CHECK_USAGE}</li>
+ *   <li>{@link OssFeatures#OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD}</li>
+ * </ul>
+ */
+public class UsesOwaspDependencyScan extends GitHubCachingDataProvider {
+
+  /**
+   * A list of configurations associated to OWASP Dependency check plugin which can fail the build.
+   */
+  private static final Map<String, String> FAIL_BUILD_CONFIGURATIONS = new HashMap<>();
+
+  static {
+    FAIL_BUILD_CONFIGURATIONS.put("junitFailOnCVSS", "number"); 
+    FAIL_BUILD_CONFIGURATIONS.put("failBuildOnCVSS", "number");
+    FAIL_BUILD_CONFIGURATIONS.put("failBuildOnAnyVulnerability", "boolean");
+  }
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param fetcher An interface to GitHub.
+   */
+  public UsesOwaspDependencyScan(GitHubDataFetcher fetcher) {
+    super(fetcher);
+  }
+
+  @Override
+  protected Set<Feature> supportedFeatures() {
+    return features();
+  }
+
+  @Override
+  protected ValueSet fetchValuesFor(GitHubProject project) throws IOException {
+    logger.info("Figuring out if the project uses OWASP Dependency Check ...");
+    LocalRepository repository = GitHubDataFetcher.localRepositoryFor(project);
+
+    ValueSet values = new ValueHashSet();
+    checkMaven(repository, values);
+    checkGradle(repository, values);
+
+    return values;
+  }
+
+  /**
+   * Returns the set of OWASP Dependency Check features.
+   */
+  static Set<Feature> features() {
```

Comment:
I am not sure if this method and `OWASP_FEATURES` are necessary. We already have `supportedFeatures()` that should return a set of supported features. Let's keep it simple unless there is a strong reason.

Reply:
You are right. I have removed this.